OctoAwesome/OctoAwesome.Runtime/GameService.cs

```diff
@@ -0,0 +1,157 @@
+        /// <returns>Geschwindigkeit der <see cref="Entity"/> nach der Killisionsprüfung</returns>
+        public Vector3 WorldCollision(GameTime gameTime, Coordinate position, ILocalChunkCache cache, float radius, float height, 
+            Vector3 deltaPosition, Vector3 velocity)
+        {
+            if (cache == null)
+                throw new ArgumentNullException(nameof(cache));
+
+            Vector3 move = deltaPosition;
+
+            //Blocks finden die eine Kollision verursachen könnten
+            int minx = (int) Math.Floor(Math.Min(
+                position.BlockPosition.X - radius,
+                position.BlockPosition.X - radius + deltaPosition.X));
+            int maxx = (int) Math.Ceiling(Math.Max(
+                position.BlockPosition.X + radius,
+                position.BlockPosition.X + radius + deltaPosition.X));
+            int miny = (int) Math.Floor(Math.Min(
+                position.BlockPosition.Y - radius,
+                position.BlockPosition.Y - radius + deltaPosition.Y));
+            int maxy = (int) Math.Ceiling(Math.Max(
+                position.BlockPosition.Y + radius,
+                position.BlockPosition.Y + radius + deltaPosition.Y));
+            int minz = (int) Math.Floor(Math.Min(
+                position.BlockPosition.Z,
+                position.BlockPosition.Z + deltaPosition.Z));
+            int maxz = (int) Math.Ceiling(Math.Max(
+                position.BlockPosition.Z + height,
+                position.BlockPosition.Z + height + deltaPosition.Z));
+
+            //Beteiligte Flächen des Spielers
+            var playerplanes = CollisionPlane.GetEntityCollisionPlanes(radius, height, velocity, position);
+
+            bool abort = false;
+
+            for (int z = minz; z <= maxz && !abort; z++)
+            {
+                for (int y = miny; y <= maxy && !abort; y++)
+                {
+                    for (int x = minx; x <= maxx && !abort; x++)
+                    {
+                        move = velocity * (float) gameTime.ElapsedGameTime.TotalSeconds;
+
+                        Index3 pos = new Index3(x, y, z);
+                        Index3 blockPos = pos + position.GlobalBlockIndex;
+                        ushort block = cache.GetBlock(blockPos);
+
+                        if (block == 0)
+                            continue;
+
+                        var blockplane = CollisionPlane.GetBlockCollisionPlanes(pos, velocity);
+
+                        var planes = from pp in playerplanes
+                                     from bp in blockplane
+                                     where CollisionPlane.Intersect(bp, pp)
+                                     let distance = CollisionPlane.GetDistance(bp, pp)
+                                     where CollisionPlane.CheckDistance(distance, move)
+                                     select new { BlockPlane = bp, PlayerPlane = pp, Distance = distance };
+
```

Comment:
Verstößt zwar auch gegen die Richtlinien, aber weil das alter Code ist, ist das in Ordnung.
Wenn du das aber dennoch umschreiben kannst oder möchtest gerne, sonst OK.

Reply:
das ist dein code :P